core/src/main/java/net/kyori/moonshine/strategy/IPlaceholderResolverStrategy.java

```diff
@@ -0,0 +1,48 @@
+//
+// moonshine - A localisation library for Java.
+// Copyright (C) 2021 Mariell Hoversholm
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+package net.kyori.moonshine.strategy;
+
+import net.kyori.moonshine.Moonshine;
+import net.kyori.moonshine.exception.PlaceholderResolvingException;
+import net.kyori.moonshine.model.MoonshineMethod;
+import java.util.Map;
+import javax.annotation.concurrent.ThreadSafe;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * A strategy for resolving placeholders in a message.
```

Comment:
This feels very vague. Again, I can take a *guess*, but a bit more detail wouldn't hurt. What should a strategy be expected to do here in relation to `IPlaceholderResolver`s?